compiler-rt/lib/asan/asan_poisoning.cpp

```diff
@@ -119,7 +181,7 @@ void __asan_poison_memory_region(void const volatile *addr, uptr size) {
       if (beg.offset > 0) {
         *beg.chunk = Min(value, beg.offset);
       } else {
-        *beg.chunk = kAsanUserPoisonedMemoryMagic;
+        *beg.chunk = poison_magic;
```

Comment:
Why can't we use kAsanUserPoisonedMemoryMagic always?

Reply:
Multiple poison values let us shard into different ring buffers. Having multiple ring buffers:

- reduces the likelihood of lost records (race conditions) when pushing onto the ring buffers
- improves the lookup speed when searching for the stack, because we can look up in one of the smaller ring buffers instead of one giant ring buffer

Reply:
1. we should have no "race conditions", though it helps with contention, but I'd prefer simpler version, which does not redefine shadow constants
2. It's on report only, perf is not important.

Reply:
Changed to single poison value